fix(pragma): PragmaState struct literals missing features field after #3544 (#3544)

## Problem

PR #3544 added a `features: Vec<&'static str>` field to `PragmaState` in `perl-pragma`, but two call sites that construct `PragmaState` with struct literal syntax were not updated. This causes `error[E0063]: missing field 'features' in initializer of PragmaState` when running `cargo test -p perl-parser`:

- `crates/perl-parser/tests/scope_analysis_test.rs:27`
- `crates/perl-parser/benches/scope_benchmark.rs:76`

Additionally, `v5_12_does_not_have_switch` in `comprehensive_unit_tests.rs` has a misleading name — it actually asserts switch IS present at v5.12 (correct behavior) but the name says "does not have".

## Fix

Add `features: Vec::new()` to both struct literals, and rename the misleading test to `v5_12_retains_switch_from_v5_10`.

## Files

- `crates/perl-parser/tests/scope_analysis_test.rs` line 27
- `crates/perl-parser/benches/scope_benchmark.rs` line 76
- `crates/perl-pragma/tests/comprehensive_unit_tests.rs` — rename test function

--- a/crates/perl-parser/benches/scope_benchmark.rs
+++ b/crates/perl-parser/benches/scope_benchmark.rs
@@ -73,7 +73,13 @@ fn benchmark_strict_barewords(c: &mut Criterion) {
     // Enable strict subs to force is_known_function checks
     let pragma_map = vec![(
         0..script.len(),
-        PragmaState { strict_subs: true, strict_vars: true, strict_refs: true, warnings: true },
+        PragmaState {
+            strict_subs: true,
+            strict_vars: true,
+            strict_refs: true,
+            warnings: true,
+            features: Vec::new(),
+        },
     )];
 
     c.bench_function("scope_analysis_strict_barewords", |b| {

--- a/crates/perl-parser/tests/scope_analysis_test.rs
+++ b/crates/perl-parser/tests/scope_analysis_test.rs
@@ -24,7 +24,13 @@ fn analyze_strict(code: &str) -> Vec<ScopeIssue> {
     // Create a pragma map with strict enabled for the whole file
     let pragma_map = vec![(
         0..code.len(),
-        PragmaState { strict_refs: true, strict_subs: true, strict_vars: true, warnings: true },
+        PragmaState {
+            strict_refs: true,
+            strict_subs: true,
+            strict_vars: true,
+            warnings: true,
+            features: Vec::new(),
+        },
     )];
 
     analyzer.analyze(&ast, code, &pragma_map)

--- a/crates/perl-pragma/tests/comprehensive_unit_tests.rs
+++ b/crates/perl-pragma/tests/comprehensive_unit_tests.rs
@@ -747,12 +747,12 @@ fn v5_40_enables_builtin() -> Result<(), Box<dyn std::error::Error>> {
 }
 
 #[test]
-fn v5_12_does_not_have_switch() -> Result<(), Box<dyn std::error::Error>> {
-    // switch was removed/deprecated in v5.38
+fn v5_12_retains_switch_from_v5_10() -> Result<(), Box<dyn std::error::Error>> {
+    // switch was inherited from v5.10 and not removed until v5.38
     let features_v12 = features_enabled_by_version(PerlVersion::new(5, 12));
     assert!(
         features_v12.contains(&"switch"),
-        "v5.12 should include 'switch' (inherited from v5.10)"
+        "v5.12 should include 'switch' (inherited from v5.10, removed at v5.38)"
     );
     Ok(())
 }

